module/remote/thriftbackend/CoreSocket.py

```diff
@@ -0,0 +1,141 @@
+        start = time.time()
+        while True:
+            try:
+                return self.__dict__["connection"].send(buff)
+            except WantReadError:
+                time.sleep(self.ssl_retry)
+            if time.time() - start > self.ssl_timeout:
+                #print "SSL connection timed out for peer: %s:%d (send)" % self.__dict__["connection"].getpeername()
+                return ''
+
+    def recv(self, buff):
+        start = time.time()
+        while True:
+            try:
+                return self.__dict__["connection"].recv(buff)
+            except WantReadError:
+                time.sleep(self.ssl_retry)
+            if time.time() - start > self.ssl_timeout:
+                #print "SSL connection timed out for peer: %s:%d (recv)" % self.__dict__["connection"].getpeername()
+                return ''
+
+class Socket(TSocket):
+    def __init__(self, host='localhost', port=7228, ssl=False):
+        TSocket.__init__(self, host, port)
+        self.ssl = ssl
+
+    def open(self):
+        if self.ssl:
+            SSL = __import__("OpenSSL", globals(), locals(), "SSL", -1).SSL
+            WantReadError = SSL.WantReadError
+            ctx = SSL.Context(SSL.SSLv23_METHOD)
+            c = SSL.Connection(ctx, socket.socket(socket.AF_INET, socket.SOCK_STREAM))
+            c.set_connect_state()
+            self.handle = SecureSocketConnection(c)
+        else:
+            self.handle = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+        #errno 104 connection reset
+
+        self.handle.settimeout(self._timeout)
+        self.handle.connect((self.host, self.port))
+
+    def read(self, sz):
+        try:
+            buff = self.handle.recv(sz)
+        except socket.error, e:
+            if (e.args[0] == errno.ECONNRESET and
+                (sys.platform == 'darwin' or sys.platform.startswith('freebsd'))):
+                # freebsd and Mach don't follow POSIX semantic of recv
+                # and fail with ECONNRESET if peer performed shutdown.
+                # See corresponding comment and code in TSocket::read()
+                # in lib/cpp/src/transport/TSocket.cpp.
+                self.close()
+                # Trigger the check to raise the END_OF_FILE exception below.
+                buff = ''
+            else:
+                raise
+        except Exception, e:
+            # SSL connection was closed
+            if e.args == (-1, 'Unexpected EOF'):
+                buff = ''
+            elif e.args == ([('SSL routines', 'SSL23_GET_CLIENT_HELLO', 'unknown protocol')],):
+                #a socket not using ssl tried to connect
+                buff = ''
+            else:
+                raise
+            
+        if not len(buff):
+            raise TTransportException(type=TTransportException.END_OF_FILE, message='TSocket read 0 bytes')
+        return buff
+
+
+class ServerSocket(TServerSocket, Socket):
+    def __init__(self, port=7228, host="0.0.0.0", key="", cert=""):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Possible binding to all interfaces.](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=567903)